projects/igniteui-angular/src/lib/tabbar/tabbar.component.ts

```diff
@@ -381,19 +462,50 @@ export class IgxTabComponent {
      * ```
      * @memberof IgxTabComponent
      */
-    @HostBinding('attr.role') public role = 'tab';
+    @HostBinding('attr.role')
+    public role = 'tab';
+
     /**
      * Gets the panel associated with the tab.
      * ```typescript
      * let tabPanel =  this.tab.relatedPanel;
      * ```
      * @memberof IgxTabComponent
      */
-    @Input() public relatedPanel: IgxTabPanelComponent;
+    @Input()
+    public relatedPanel: IgxTabPanelComponent;
+
+    /**
+     * Sets/gets the `label` of the tab panel.
+     * ```html
+     * <igx-tab [label] = "'Tab label'"><igx-tab>
+     * ```
+     * ```typescript
+     * let tabLabel = this.tab.label;
+     * ```
+     * @memberof IgxTabComponent
+     */
+    @Input()
+    public label: string;
```

Comment:
Consider getting/setting the relatedPanel.label and relatedPanel.icon when the tabs have relatedPanel. This will make the context bindings in the template unnecessary.

Reply:
Addressed.